cupy/core/_routines_statistics.pyx

```diff
@@ -89,6 +89,9 @@ cdef ndarray _ndarray_argmax(ndarray self, axis, out, dtype, keepdims):
     for accelerator in _accelerator._routine_accelerators:
         if accelerator == _accelerator.ACCELERATOR_CUB:
             # result will be None if the reduction is not compatible with CUB
+            if self._f_contiguous and self.dtype == numpy.bool_:
+                # temporary workaround
+                self = self.astype(numpy.int8)
```

Comment:
CUB `argmax` seems to return different values to NumPy for F-order bool array inputs. In this PR, I made a temporary fix casting the inputs to int8 data type. (cc: @leofang)